skimage/filters/_gaussian.py

```diff
@@ -129,6 +129,12 @@ def gaussian(image, sigma=1, output=None, mode='nearest', cval=0,
             sigma = [sigma] * (image.ndim - 1)
         if len(sigma) != image.ndim:
             sigma = np.concatenate((np.asarray(sigma), [0]))
+    if output is not None and not preserve_range:
+        if isinstance(output, np.ndarray):
+            if issubclass(output.dtype.type, np.integer):
+                preserve_range = True
+        elif issubclass(output, np.integer):
```

Comment:
oh my god, this is so confusing. The fact that output can be both a dtype, or an array is killer.

Reply:
I didn't get why `skimage.filters.gaussian` converting all type images? Is `convert_to_float` really needed for all dtypes? `scipy.ndimage.gaussian_filter` works correctly with all numpy dtypes

Reply:
because images in uint8 format are between 0 and 255 inclusive, but "images" in float32 (and float64) are between 0.0 and 1.0 inclusive.

The only thing that is different is thus the scaling factor. Annoying, but it is pretty standard in most "image" formats.

Reply:
@snasibli, are you expecting to finish this PR? I could help out if you're too busy. I'm new to scikit-image but I already found those data types inconsistencies very confusing. Shouldn't there be some common checks for all functions that expect specific types? If every filter has its own type-checking it will end up a big mess, I'm worried.

Reply:
It is needed to understand what it is output.  Is it numpy dtype or allocateed numpy array, or both? In docstring of skimage.filters.gaussian output is expected as an array, but for scipy.ndimage.gaussian_filter it could be as dtype as array.
I agree with that there should be some common checks for all functions that expect specific types. If I will get answers to this questions I finish

Reply:
@hmaarrfk, @mixmixmix can you please, ask the questions on previous comment. I'm going to finish this work)

Reply:
The fact that output can be a dtype or array seems like a strange convention in numpy/scipy functions.. i guess we should follow.